dbms/src/Storages/Page/WriteBatchWrapper.h

```diff
@@ -0,0 +1,160 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <IO/ReadBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <Storages/Page/PageDefinesBase.h>
+#include <Storages/Page/V3/Universal/UniversalWriteBatchAdaptor.h>
+#include <Storages/Page/WriteBatch.h>
+#include <fmt/format.h>
+
+#include <vector>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int LOGICAL_ERROR;
+} // namespace ErrorCodes
+
+enum class PageStorageRunMode : UInt8
+{
+    ONLY_V2 = 1,
+    ONLY_V3 = 2,
+    MIX_MODE = 3,
+    UNI_PS = 4,
+};
+
+// It contains either an UniversalWriteBatchAdaptor or a WriteBatch.
+class WriteBatchWrapper : private boost::noncopyable
```

Comment:
Since this class is used everywhere, can we just name it `WriteBatch`?

Reply:
`WriteBatch` is rarely used in the code. Most of its' use is through WriteBatches. So it only introduces really small changes in our code.
And `WriteBatch` is mostly used in the test code. if we rename `WriteBatchWrapper` to `WriteBatch`, the constructor of "`WriteBatch`" will change. This will introduce a lot of unnecessary change in the test code.

Reply:
How about make `WriteBatch` a abstract class, and implement `WriteBatchU64` and `WriteBatchString`. Then using  `WriteBatchPtr` as parameters and pass around.

Reply:
`WriteBatchU64` and `WriteBatchUniversal` shares really little interface.
Because proxy data must write its data using `WriteBatchUniversal::putPage(String, ...)`. There is no prefix for proxy data.
And we don't need a `WriteBatchPtr` to be passed around. All changes are enclosed in `WriteBatches`.